homeassistant/components/device_tracker/icloud.py

```diff
@@ -430,7 +430,8 @@ def lost_iphone(self, devicename):
         self.api.authenticate()
 
         for device in self.api.devices:
-            if devicename is None or device == self.devices[devicename]:
+            if (devicename is None or
+                    str(device) == str(self.devices[devicename])):
```

Comment:
Casting both to a string seems very weird. Which types are `device` and `self.devices[devicename]` ? Why is this working ?

Reply:
I don't know actually. I just copied this from a PR in the original repo if this component

Reply:
Well that's not good enough reason… we can only accept proper fixes.

Reply:
I don't know/don't have time to setup a developer instance of HA, but the comments on the original PR were positive so I'm confident it's going to fix it.